test/routes/api/tasks/tasks.test.ts

```diff
@@ -457,182 +471,277 @@ describe('Tasks api (logged user only)', () => {
-      const res = await app.injectWithLogin('basic', {
-        method: 'GET',
-        url: `/api/tasks/${taskFilename}/image`
+        await app.knex<Task>('tasks').where({ id: taskId }).update({ filename: null})
+        
+        const files = fs.readdirSync(uploadDirTask)
+        files.forEach((file) => {
+          const filePath = path.join(uploadDirTask, file)
+          fs.rmSync(filePath, { recursive: true })
+        })       
+
+        await app.close()
       })
 
-      assert.strictEqual(res.statusCode, 200)
-      assert.strictEqual(res.headers['content-type'], 'image/png')
+      beforeEach(async () => {
+        app = await build()
+        await uploadImageForTask(app, taskId, testImagePath, uploadDirTask)
+        await app.close()
+      })
 
-      const originalFile = fs.readFileSync(testImagePath)
+      after(async () => {
+        app = await build()
+        const files = fs.readdirSync(uploadDirTask)
+        files.forEach((file) => {
+          const filePath = path.join(uploadDirTask, file)
+          fs.rmSync(filePath, { recursive: true })
+        })
 
-      assert.deepStrictEqual(originalFile, res.rawPayload)
-    })
+        await app.close()
+      })
 
-    it('should return 404 error for non-existant filename', async (t) => {
-      app = await build(t)
+      it('should remove an existing image for a task', async (t) => {
+        app = await build(t)
 
-      const res = await app.injectWithLogin('basic', {
-        method: 'GET',
-        url: '/api/tasks/non-existant/image'
+        const taskFilename = encodeURIComponent(`${taskId}_${filename}`)
+        const resDelete = await app.injectWithLogin('basic', {
+          method: 'DELETE',
+          url: `/api/tasks/${taskFilename}/image`
+        })
+
+        assert.strictEqual(resDelete.statusCode, 204)
+
+        const files = fs.readdirSync(uploadDirTask)
+        assert.strictEqual(files.length, 0)
       })
 
-      assert.strictEqual(res.statusCode, 404)
-      const { message } = JSON.parse(res.payload)
-      assert.strictEqual(message, 'No task has filename "non-existant"')
+      it('should return 404 for non-existant filename for deletion', async (t) => {
+        app = await build(t)
+
+        const res = await app.injectWithLogin('basic', {
+          method: 'DELETE',
+          url: '/api/tasks/non-existant/image'
+        })
+
+        assert.strictEqual(res.statusCode, 404)
+        const { message } = JSON.parse(res.payload)
+        assert.strictEqual(message, 'No task has filename "non-existant"')
+      })
+
+      it('File deletion transaction should rollback on error', async (t) => {
+        const app = await build(t)
+        const { mock: mockPipeline } = t.mock.method(fs.promises, 'unlink')
+        mockPipeline.mockImplementationOnce(() => {
+          throw new Error()
+        })
+
+        const { mock: mockLogError } = t.mock.method(app.log, 'error')
```

Comment:
I think you can just save a task with a filename that does not exist in upload directory and it will trigger a `ENOENT` error, ensuring the transaction rollback.

Reply:
https://github.com/fastify/demo/pull/46#discussion_r1818070252

Reply:
You'll have to throw an other error than `ENOENT`, so you can keep the mock.